src/OpenConext/ProfileBundle/Controller/MyServicesController.php

```diff
@@ -31,38 +33,42 @@ class MyServicesController
     private $templateEngine;
 
     /**
-     * @var TokenStorageInterface
+     * @var SpecifiedConsentListService
      */
-    private $tokenStorage;
+    private $specifiedConsentListService;
 
     /**
-     * @var ConsentService
+     * @var UserProvider
      */
-    private $consentService;
+    private $userProvider;
 
     /**
      * @param EngineInterface $templateEngine
-     * @param TokenStorageInterface $tokenStorage
-     * @param ConsentService $consentService
+     * @param UserProvider $userProvider
+     * @param SpecifiedConsentListService $specifiedConsentListService
      */
     public function __construct(
         EngineInterface $templateEngine,
-        TokenStorageInterface $tokenStorage,
-        ConsentService $consentService
+        UserProvider $userProvider,
+        SpecifiedConsentListService $specifiedConsentListService
     ) {
-        $this->templateEngine = $templateEngine;
-        $this->tokenStorage   = $tokenStorage;
-        $this->consentService = $consentService;
+        $this->templateEngine              = $templateEngine;
+        $this->userProvider                = $userProvider;
+        $this->specifiedConsentListService = $specifiedConsentListService;
     }
 
     public function overviewAction()
     {
-        $user        = $this->tokenStorage->getToken()->getUser();
-        $consentList = $this->consentService->findAllFor($user);
+        if (!$this->userProvider->hasCurrentUser()) {
+            throw new RuntimeException('Cannot retrieve MyServices overview: no user available');
```

Comment:
This should not be done, rather a guard should be added that this method may only be invoked when there is a user logged in, which we enforce through the security config and verify here (see [the relevant issue][1]). I'd rather put in a comment with a todo than this tbh.

[1]: https://www.pivotaltracker.com/story/show/107020530